jlm/mlir/frontend/MlirToRvsdg.cpp

```diff
@@ -0,0 +1,256 @@
+      {
+        // If there is no defining op, the Value is necessarily a Block argument.
+        auto blockArg = operand.cast<mlir::BlockArgument>();
+        inputs.push_back(rvsdgRegion.argument(blockArg.getArgNumber()));
+      }
+    }
+
+    if (rvsdg::node * node = convertOperation(mlirOp, rvsdgRegion, inputs))
+    {
+      operations[&mlirOp] = node;
+    }
+  }
+
+  // The results of the region/block are encoded in the terminator operation
+  auto terminator = block.getTerminator();
+  std::unique_ptr<std::vector<jlm::rvsdg::output *>> results =
+      std::make_unique<std::vector<jlm::rvsdg::output *>>();
+  for (mlir::Value operand : terminator->getOperands())
+  {
+    if (mlir::Operation * producer = operand.getDefiningOp())
+    {
+      size_t index = getOperandIndex(producer, operand);
+      results->push_back(operations[producer]->output(index));
+    }
+    else
+    {
+      // If there is no defining op, the Value is necessarily a Block argument.
+      auto blockArg = operand.cast<mlir::BlockArgument>();
+      results->push_back(rvsdgRegion.argument(blockArg.getArgNumber()));
+    }
+  }
+
+  return results;
+}
+
+rvsdg::node *
+MlirToRvsdg::convertOperation(
+    mlir::Operation & mlirOperation,
+    rvsdg::region & rvsdgRegion,
+    std::vector<const rvsdg::output *> & inputs)
+{
+  if (mlirOperation.getName().getStringRef() == mlir::rvsdg::OmegaNode::getOperationName())
+  {
+    convertOmega(mlirOperation, rvsdgRegion);
+    // Omega doesn't have a corresponding RVSDG node so we return nullptr
+    return nullptr;
+  }
+  else if (mlirOperation.getName().getStringRef() == mlir::rvsdg::LambdaNode::getOperationName())
+  {
+    return convertLambda(mlirOperation, rvsdgRegion);
+  }
+  else if (mlirOperation.getName().getStringRef() == mlir::arith::ConstantIntOp::getOperationName())
+  {
+    auto constant = static_cast<mlir::arith::ConstantIntOp>(&mlirOperation);
+    auto type = constant.getType();
+    JLM_ASSERT(type.getTypeID() == mlir::IntegerType::getTypeID());
+    auto * integerType = static_cast<mlir::IntegerType *>(&type);
+
+    return rvsdg::node_output::node(
+        rvsdg::create_bitconstant(&rvsdgRegion, integerType->getWidth(), constant.value()));
+  }
+  else if (mlirOperation.getName().getStringRef() == mlir::rvsdg::LambdaResult::getOperationName())
+  {
+    // This is a terminating operation that doesn't have a corresponding RVSDG node
+    return nullptr;
+  }
+  else if (mlirOperation.getName().getStringRef() == mlir::rvsdg::OmegaResult::getOperationName())
+  {
+    // This is a terminating operation that doesn't have a corresponding RVSDG node
+    return nullptr;
+  }
+  else
+  {
+    throw util::error(
+        "Operation is not implemented:" + mlirOperation.getName().getStringRef().str() + "\n");
+  }
+}
+
+void
+MlirToRvsdg::convertOmega(mlir::Operation & mlirOmega, rvsdg::region & rvsdgRegion)
+{
+  // The Omega has a single region
+  JLM_ASSERT(mlirOmega.getRegions().size() == 1);
```

Comment:
The comment is superfluous. It only states the same as the `JLM_ASSERT`.

Reply:
Removed